File: collect_app/src/main/java/org/odk/collect/android/views/MultiClickSafeImageButton.java
package org.odk.collect.android.views;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageButton;

import org.odk.collect.android.utilities.MultiClickGuard;

public class MultiClickSafeImageButton extends AppCompatImageButton {
    public MultiClickSafeImageButton(@NonNull Context context) {
        super(context);
    }

    public MultiClickSafeImageButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    public boolean performClick() {
        return MultiClickGuard.allowClick(getClass().getName()) && super.performClick();
    }
}

File: collect_app/src/main/java/org/odk/collect/android/views/MultiClickSafeImageButton.kt
package org.odk.collect.android.views

import android.content.Context
import android.util.AttributeSet
import org.odk.collect.android.utilities.MultiClickGuard.allowClick
import androidx.appcompat.widget.AppCompatImageButton

class MultiClickSafeImageButton : AppCompatImageButton {
    constructor(context: Context) : super(context)

    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    )

    override fun performClick(): Boolean {
        return allowClick() && super.performClick()
    }
}
